metaflow/datatools/s3.py

```diff
@@ -1348,32 +1368,245 @@ def _s3op_with_retries(self, mode, **options):
         if self._s3_client_params is not None:
             cmdline.extend(("--s3clientparams", json.dumps(self._s3_client_params)))
 
-        for i in range(S3_RETRY_COUNT + 1):
+        def _get_inject_failures():
+            # list mode does not do retries on transient failures (there is no
```

Comment:
list does not deal with transient failures. It was too complicated to add iirc and I don't think it is worth it. The rationale for capping at 90 (and not 100) is that you could otherwise fail to make progress and have more flaky tests (there is a cap on the number of full retries and no progress counts as making a full retry).